bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -471,6 +477,7 @@ void PreProcessor::onMessage<ClientBatchRequestMsg>(ClientBatchRequestMsg *msg)
 // Non-primary replica request handling
 template <>
 void PreProcessor::onMessage<PreProcessRequestMsg>(PreProcessRequestMsg *msg) {
+  concord::diagnostics::TimeRecorder scoped_timer(*histograms_.onPreProcessRequestMsg);
```

Comment:
Is this histogram updated from multiple threads? If so, this should be changed to use the atomic variant with `TimeRecorder<true>`.

Reply:
It is always called from IncomingMsgsStorage context, which is alone. Thanks for the reminder!

Reply:
Cool. Thanks :)